src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.Entry.Roundtrip.Tests.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace System.Formats.Tar.Tests
+{
+    public class TarWriter_WriteEntry_Roundtrip_Tests : TarTestsBase
+    {
+        public static IEnumerable<object[]> NameRoundtripsTheoryData()
+        {
+            foreach (TarEntryType entryType in new[] { TarEntryType.RegularFile, TarEntryType.Directory })
+            {
+                foreach (string name in GetNamesNonAsciiTestData(NameCapabilities.Name).Concat(GetNamesPrefixedTestData(NameCapabilities.Name)))
+                {
+                    TarEntryType v7EntryType = entryType is TarEntryType.RegularFile ? TarEntryType.V7RegularFile : entryType;
+                    yield return new object[] { TarEntryFormat.V7, v7EntryType, name };
+                }
+
+                foreach (string name in GetNamesNonAsciiTestData(NameCapabilities.NameAndPrefix).Concat(GetNamesPrefixedTestData(NameCapabilities.NameAndPrefix)))
+                {
+                    yield return new object[] { TarEntryFormat.Ustar, entryType, name };
+                }
+
+                foreach (string name in GetNamesNonAsciiTestData(NameCapabilities.Unlimited).Concat(GetNamesPrefixedTestData(NameCapabilities.Unlimited)))
+                {
+                    yield return new object[] { TarEntryFormat.Pax, entryType, name };
+                    yield return new object[] { TarEntryFormat.Gnu, entryType, name };
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(NameRoundtripsTheoryData))]
+        public void NameRoundtrips(TarEntryFormat entryFormat, TarEntryType entryType, string name)
+        {
+            TarEntry entry = InvokeTarEntryCreationConstructor(entryFormat, entryType, name);
+            entry.Name = name;
+
+            MemoryStream ms = new();
```

Comment:
Will have to extend this test in [my PR for the extended attributes fix](https://github.com/dotnet/runtime/pull/75237) to include unseekable archives, because name gets written in the data section (extended attribute). The test might fail in this PR.

Reply:
I see you added them and they are not throwing. Nice.